File: base64_decode_tool.py
from pathlib import Path
import base64
from check_file_permission import check_file_permission

def base64_decode(file_path, output_path, config):
    if not check_file_permission(file_path, "read", config) or not check_file_permission(output_path, "write", config):
        raise PermissionError("Access denied for input or output")
    file_path, output_path = Path(file_path).resolve(), Path(output_path).resolve()
    with open(file_path, "r", encoding="utf-8") as f:
        content = base64.b64decode(f.read())
    with open(output_path, "wb") as f:
        f.write(content)
    return f"Decoded to {output_path}"

File: ts-tools/base64_decode_tool.ts
/**
 * @license
 * Copyright 2025 Google LLC
 * SPDX-License-Identifier: Apache-2.0
 */

import * as fs from 'fs';
import * as path from 'path';
import { checkFilePermission } from './check_file_permission';

export function base64Decode(
  filePath: string,
  outputPath: string,
  config: any,
): Promise<string> {
  return new Promise((resolve, reject) => {
    Promise.all([
      checkFilePermission(filePath, 'read', config),
      checkFilePermission(outputPath, 'write', config),
    ])
      .then(() => {
        const resolvedPath = path.resolve(filePath);
        const resolvedOutPath = path.resolve(outputPath);
        fs.readFile(resolvedPath, 'utf-8', (err, data) => {
          if (err) {
            reject(err);
          } else {
            const decoded = Buffer.from(data, 'base64');
            fs.writeFile(resolvedOutPath, decoded, (err) => {
              if (err) {
                reject(err);
              } else {
                resolve(`Decoded to ${outputPath}`);
              }
            });
          }
        });
      })
      .catch(reject);
  });
}
